### PR TITLE
Distribution mapping and cost plotting

### PR DESCRIPTION
**Pull request by mrowan137**
This PR adds scripts for visualizing distribution mappings and costs produced with load balance reduced diagnostics.

### CODE REVIEW

**Comment by ax3l** [comment]
@mrowan137 sorry for being so slow in the review.
Can you please give an example how to use this for a given warpx inputs file? Do we also need to activate a specific reduced diagnostics?

cc @guj

**Comment by mrowan137** [comment]
@ax3l 
> Can you please give an example how to use this for a given warpx inputs file? Do we also need to activate a specific reduced diagnostics?

right, the user needs to specify `LoadBalanceCosts` reduced diagnostic.  maybe I can add this as a comment in the Jupyter notebook?  or include a full input file/submit script in the example directory?  the relevant lines of input are:
```
warpx.reduced_diags_names = LBC                          
LBC.type = LoadBalanceCosts 
LBC.frequency = 100
```

**Comment by ax3l** [comment]
re-triggering CI.

**Comment by ax3l** [comment]
@guj did the plotting work for you two for 2D & 3D? :)
Should we add a note on how to create this in the docs?

File: Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
...5]j_low_box_56() [456]k_low_box_56() [457]gpu_ID_box_56() [458]hostname_box_56() [459]cost_box_57() [460]proc_box_57() [461]lev_box_57() [462]i_low_box_57() [463]j_low_box_57() [464]k_low_box_57() [465]gpu_ID_box_57() [466]hostname_box_57() [467]cost_box_58() [468]proc_box_58() [469]lev_box_58() [470]i_low_box_58() [471]j_low_box_58() [472]k_low_box_58() [473]gpu_ID_box_58() [474]hostname_box_58() [475]cost_box_59() [476]proc_box_59() [477]lev_box_59() [478]i_low_box_59() [479]j_low_box_59() [480]k_low_box_59() [481]gpu_ID_box_59() [482]hostname_box_59() [483]cost_box_60() [484]proc_box_60() [485]lev_box_60() [486]i_low_box_60() [487]j_low_box_60() [488]k_low_box_60() [489]gpu_ID_box_60() [490]hostname_box_60() [491]cost_box_61() [492]proc_box_61() [493]lev_box_61() [494]i_low_box_61() [495]j_low_box_61() [496]k_low_box_61() [497]gpu_ID_box_61() [498]hostname_box_61() [499]cost_box_62() [500]proc_box_62() [501]lev_box_62() [502]i_low_box_62() [503]j_low_box_62() [504]k_low_box_62() [505]gpu_ID_box_62() [506]hostname_box_62() [507]cost_box_63() [508]proc_box_63() [509]lev_box_63() [510]i_low_box_63() [511]j_low_box_63() [512]k_low_box_63() [513]gpu_ID_box_63() [514]hostname_box_63()
```
Comment (ax3l) [inline Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt]: Should we check in those data files? We probably don't need them, do we? Instead let us document this workflow maybe?

Comment (mrowan137) [reply Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt]: I guess they aren't super-important, but they do allow someone to use the plotting scripts out-of-the box without generating data themselves.  if we want to reduce bloat I can remove the samples files and tutorial notebook (which does not make sense to me without ready-to-use data).  pictures could still be included in the documentation.

**Comment by guj** [comment]
I tried 3D with a small test and it was working.

The Jupyter notebook created is visually helpful.  If possible, it would be
great to have a documentation page on how to use this feature, and what to
expect.
- the specific lines needed in input file (load balancing + reduced_diag)
- what the python scripts expect, and what are the graphs generated.

- Junmin

File: Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt (hunk 1 above)
Comment (mrowan137) [reply Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt]: @ax3l @guj any opinion on the above?

Comment (ax3l) [reply Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt]: I think let's remove those data files and add a small section in the docs to describe the workflow on how to create the plots and data for it :) is that ok @mrowan137? :)

Comment (mrowan137) [reply Tools/PostProcessing/plot_distribution_mapping/data/LBC_3D.txt]: sounds good!

**Review by ax3l — APPROVED** [review]
Thank you, great work! :rocket: :sparkles:

**Pull request merged**